include/exadg/functions_and_boundary_conditions/boundary_face_integrator_base.h

```diff
@@ -0,0 +1,84 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef EXADG_FUNCTIONS_AND_BOUNDARY_CONDITIONS_BOUNDARY_FACE_INTEGRATOR_BASE_H_
+#define EXADG_FUNCTIONS_AND_BOUNDARY_CONDITIONS_BOUNDARY_FACE_INTEGRATOR_BASE_H_
+
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <exadg/utilities/enum_utilities.h>
+
+namespace ExaDG
+{
+/**
+ * Base class to access boundary values with a similar interface to FaceIntegrators.
+ * In the derived class a @c get_value() has to be implemented. Since the return type
+ * depends on the derived class, a pure virtual function for this purpose can not be added.
```

Comment:
I think we can still improve this description. According to my understanding, the main purpose of this class is to allow a simple call `face_integrator.get_value(q)` with just the quadrature point index `q` as argument(s). To enable this, we need a class that receives the relevant parameters ({MatrixFree, BoundaryDescriptor} and {face_id, time}) during construction and `reinit()`, respectively.

It is not mandatory for this PR and I can also change it later.

Other operators might also need the (normal) gradient on boundary faces. This is another reason why such a base class is useful in my opinion (it can be used equally for first-derivative and second-derivative operators).